src/Microsoft.AspNet.Mvc.Core/Rendering/Html/HtmlHelper.cs

```diff
@@ -1016,5 +1045,22 @@ protected virtual string GenerateValue(string expression, object value, string f
         {
             return _htmlGenerator.GetClientValidationRules(ViewContext, modelExplorer, expression);
         }
+
+        protected virtual IEnumerable<SelectListItem> GetSelectList([NotNull] ModelMetadata metadata)
```

Comment:
protected is like public, should we validate here? or perhaps name this something like `GetEnumSelectList`?